File: projekty/PO_Projekt3/mlecz.py
from roslina import Roslina

class Mlecz(Roslina):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 0, 0, 0.10, 3)

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, Mlecz)

    def stworzDziecko(self, cel):
        return Mlecz(self._swiat, cel)

    def rysowanie(self):
        return (255, 255, 102)

File: PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Rosliny/Mlecz.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Rosliny;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;

public class Mlecz extends Roslina {

	public Mlecz(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(Mlecz.class), Stale.inicjatywa.get(Mlecz.class),
				Stale.szansaRozprzestrzeniania.get(Mlecz.class), Stale.probyRozprzestrzeniania.get(Mlecz.class));
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof Mlecz;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new Mlecz(swiat, cel);
	}

}
